sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveSessionCatalog.scala

```diff
@@ -66,10 +66,11 @@ private[sql] class HiveSessionCatalog(
       name: String,
       clazz: Class[_],
       input: Seq[Expression]): Expression = {
-
-    Try(super.makeFunctionExpression(name, clazz, input)).getOrElse {
-      var udfExpr: Option[Expression] = None
-      try {
+    val originalClassLoader = Thread.currentThread().getContextClassLoader()
```

Comment:
It doesn't need to modify original code but just switch context classloader. Then, use `Utils.withContextClassLoader` and just pass original code into fn.